fl4health/utils/partitioners.py

```diff
@@ -90,15 +101,21 @@ def partition_label_indices(
         """
         if self.prior_distribution is not None:
             assert (
-                len(self.prior_distribution) == self.n_unique_labels
-            ), "The length of the prior must match the number of labels"
-            partition_allocations = self.prior_distribution[label]
+                len(self.prior_distribution[label]) == self.number_of_partitions
+            ), f"The length of the prior distribution for label ({str(label)}) must match the number of partitions"
+            if sum(self.prior_distribution[label]) != 1:
+                log(
+                    WARN,
+                    f"The provided prior distribution for label ({str(label)}) does not sum to 1. "
+                    "It will be normalized to sum to 1.",
+                )
+            partition_allocations = self.prior_distribution[label] / sum(self.prior_distribution[label])
             log(
                 INFO,
-                f"Using a prior distribution of {self.prior_distribution[label]} to allocate the label ({str(label)})",
+                f"The allocation distribution for label ({str(label)}) is {partition_allocations} "
+                f"using the provided prior distribution",
```

Comment:
I'm actually not sure why flake doesn't complain about this, but most of the time it gets fussy when using an f-string without an actual variable to dereference. Maybe drop the f here to be safe?